website/src/docs/hotchocolate/defining-a-schema/extending-types.md

```diff
@@ -4,190 +4,220 @@ title: "Extending Types"
+public class Startup
 {
-    public string GetBar() => ...
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services
+            .AddGraphQLServer()
+            .AddTypeExtension<BookExtensions>();
+    }
 }
 ```
 
+</ExampleTabs.Annotation>
+<ExampleTabs.Code>
+
+TODO
+
 </ExampleTabs.Code>
 <ExampleTabs.Schema>
 
-```graphql
-type Query {
-  foo: String
-}
-
-extend type Query {
-  bar: String
-}
-```
+Simply remove the field from the existing type.
 
 </ExampleTabs.Schema>
 </ExampleTabs>
 
-> Note: Type extensions need to be registered with the GraphQL configuration. If you are using ASP.NET core head over to your `Startup.cs` and add the type extension with `AddTypeExtension` to your schema.
->
-> ```csharp
-> services
->    .AddGraphQLServer()
->    .AddQueryType<Query>()
->    .AddTypeExtension<QueryExtensions>();
-> ```
+## Replacing fields
 
-# Extending types with the annotation-based approach
+We might have an `Id` field, which we want to replace with a field that resolves the actual type the `Id` is pointing to.
 
-Extending types can be beneficial even with non-root types. Let's say we are building a schema with the annotation-based approach where we use pure C# to describe our types.
+In this example we replace the `authorId` field with an `author` field.
 
-Given is the following entity that we do want to extend in our graph with additional fields.
+<ExampleTabs>
+<ExampleTabs.Annotation>
 
 ```csharp
-public class Session
+[ExtendObjectType(typeof(Book))]
+public class BookExtensions
 {
-    public int Id { get; set; }
-
-    [Required]
-    [StringLength(200)]
-    public string? Title { get; set; }
-
-    [StringLength(4000)]
-    public string? Abstract { get; set; }
-
-    public int? TrackId { get; set; }
+    [BindMember(nameof(Book.AuthorId))]
+    public Author GetAuthor([Parent] Book book)
+    {
+        // Omitted code for brevity
+    }
 }
-```
-
-## Replace a field
-
-We could start adding our GraphQL concerns to this type directly. But often, we want to keep our entity clean from any graph concerns.
 
-To replace the `TrackId` with a field `Track` that returns the `Tack` object we could do the following.
-
-```csharp
-[ExtendObjectType(typeof(Session))]
-public class SessionResolvers
+public class Startup
 {
-    [BindMember(nameof(Session.TrackId))]
-    public async Task<Track> GetTrackAsync([Parent] Session session) => ...
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services
+            .AddGraphQLServer()
+            .AddTypeExtension<BookExtensions>();
+    }
 }
 ```
 
-## Remove a field
-
-We also easily can remove properties that we do not like on our initial type. For instance, let us omit the `Abstract`.
+</ExampleTabs.Annotation>
+<ExampleTabs.Code>
 
-```csharp
-[ExtendObjectType(
-    typeof(Session),
-    IgnoreProperties = new[] { nameof(Session.Abstract) })]
-public class SessionResolvers
-{
-}
-```
+TODO
```

Comment:
Code-first replacing existing field on existing type

Reply:
Doing this by targeting a base type is super powerful and something that'd be great to document with some real world examples. 

Personally we use this heavily in return payload types and just normal types to pull in things by just their Id. Meaning you get CQRS for free on your payloads and stitching becomes a breeze. We extend a baste type like a `IHasBlogPost` or `ICanHaveBlogPost` with the resolver to look it up by id, and using the dataloader, and then annotate all kinds of things with that to do local stitching (e.g. when not remote stitching) and the Q in CQRS, because our mutation can just set the ids on the return type and then the resolver will do a fresh fetch of it for us. 

If you need some examples, here's the original issue, though Michael's comment has all the finalised syntax really.
https://github.com/ChilliCream/hotchocolate/issues/2991